vagrant/scripts/build_rust_rpms.sh

```diff
@@ -1,21 +1,36 @@
+set -x
 yum install -y openssl-devel
-yum install -y cargo rpm-build
+yum install -y cargo rpm-build git
 
-[ -f /root/.cargo/bin/sccache ] && echo "sccache already installed. Skipping." || cargo install sccache
+unset RUSTC_WRAPPER
 
+[ -f /sccache/target/release/sccache ] \
+    && echo "sccache already installed. Skipping." \
+    || (cd / \
+        && rm -rf /sccache \
+        && git clone https://github.com/mozilla/sccache.git \
+        && cd /sccache \
+        && cargo build --release --no-default-features)
+
+sed -i '/^PATH=\$PATH/d' /root/.bash_profile
+sed -i '/^export PATH/d' /root/.bash_profile
 sed -i '/^export CARGO_HOME/d' /root/.bash_profile
 sed -i '/^export SCCACHE_CACHE_SIZE/d' /root/.bash_profile
 sed -i '/^export SCCACHE_DIR/d' /root/.bash_profile
 sed -i '/^export RUSTC_WRAPPER/d' /root/.bash_profile
 
 cat <<EOF >> /root/.bash_profile
+PATH=$PATH:$HOME/bin
+export PATH
 export CARGO_HOME=/root/.cargo
 export SCCACHE_CACHE_SIZE=40G
 export SCCACHE_DIR=/root/.cache
-export RUSTC_WRAPPER=/root/.cargo/bin/sccache
+export RUSTC_WRAPPER=/sccache/target/release/sccache
 EOF
 
 source /root/.bash_profile
 
+
 cd /integrated-manager-for-lustre \
+    && [ "$1" = "--clean" ] && cargo clean || echo "Attempting to build rpm's without cleaning." \
```

Comment:
Does it clean `sscache` as well? This seems a bit unexpected, since cargo doesn't know about sscache (or does it?).

Reply:
In my case, I made a change to one of the migrations and changed a small piece of code in one of the services. When building the rpm's, cargo just saw that that one service was changed, so when it built the rpms, it used the cache data for the other services. This ultimately led to the other services having the previous SQL migrations baked into them while the service I made a change to baked in the new migration script. So when I installed the rpms, one service had a different migration baked into it, which created a race condition to run the SQL migrations and ultimately caused the services to crash. What this does is clean out cargo to ensure the other services are also built. Sccache will still be used to speed up compile time.